agent/consul/state/catalog_test.go

```diff
@@ -1950,31 +1939,40 @@ func TestStateStore_ServicesByNodeMeta(t *testing.T) {
 		if err != nil {
 			t.Fatalf("err: %s", err)
 		}
-		expected := structs.Services{
-			"redis": []string{"primary", "prod"},
-		}
-		sort.Strings(res["redis"])
-		require.Equal(t, expected, res)
+		require.Len(t, res, 1)
+		require.True(t, res[0].ServiceID == "service1" && res[0].Node == "node0")
 	})
 
 	t.Run("Get all services using the common meta value", func(t *testing.T) {
 		_, res, err := s.ServicesByNodeMeta(ws, map[string]string{"common": "1"}, nil)
 		if err != nil {
 			t.Fatalf("err: %s", err)
 		}
-		expected := structs.Services{
-			"redis": []string{"primary", "prod", "replica"},
+		require.Len(t, res, 2)
+		expected := structs.ServiceNodes{
+			{ServiceID: "service1", Node: "node0"},
+			{ServiceID: "service1", Node: "node1"},
+		}
+		for _, sn := range res {
+			require.Condition(t, func() bool {
+				for _, exp := range expected {
+					if exp.ServiceID == sn.ServiceID && exp.Node == sn.Node {
+						return true
+					}
+				}
+				return false
+			})
 		}
```

Comment:
Similarly here, would this work?
```suggestion
		require.Equal(t, res, []*structs.NodeServbice{ns1. ns2})
```

Reply:
Same as before, this is the wrong struct type and I'm not sure how to handle that.